src/methods.jl

```diff
@@ -239,8 +239,8 @@ function fdm(
 
     p = m.p
     q = m.q
-    grid = m.grid
-    coefs = m.coefs
+    grid = T.(m.grid)
+    coefs = T.(m.coefs)
```

Comment:
This will allocate.
Can we avoid allocating by moving the conversion into the `sum` ?
So it only is converting 1 element at a time, and thus can do it on the stack and not allocate?